rust/main/chains/hyperlane-aleo/src/types.rs

```diff
@@ -11,4 +19,184 @@ use snarkvm_console_account::Field;
 // The Block type verifies its validity on creation and that changes based on the Network type, that's why we have to pass the correct Type when dealing with blocks.
 pub(crate) type CurrentNetwork = MainnetV0;
 /// TxID Type
-pub(crate) type TxID = AleoID<Field<TestnetV0>, { hrp2!("at") }>;
+pub(crate) type TxID = AleoID<Field<CurrentNetwork>, { hrp2!("at") }>;
+
+// The aleo credits have 6 decimals
+const ALEO_CREDITS_DECIMALS: u32 = 6;
+
+/// Aleo Merkle Tree
+#[aleo_serialize]
+#[derive(Debug)]
+pub struct AleoMerkleTree {
+    /// Leaf Branch
+    /// Each leaf is 32Bytes encoded as [u128; 2], where the u128 is little endian encoded
+    pub branch: [[u128; 2]; 32],
+    /// Number of inserted elements
+    pub count: u32,
+}
+
+impl From<AleoMerkleTree> for IncrementalMerkle {
+    fn from(val: AleoMerkleTree) -> Self {
+        let branch = val.branch.map(|hash| u128_to_hash(&hash));
+        IncrementalMerkle {
+            branch,
+            count: val.count as usize,
+        }
+    }
+}
+
+/// Aleo Merkle Tree Hook
+#[aleo_serialize]
+#[derive(Debug)]
+pub struct AleoMerkleTreeHookStruct {
+    /// Merkle Tree
+    pub tree: AleoMerkleTree,
+    /// Computed on chain merkle root as [u128; 2]
+    /// u128 is little endian encoded
+    pub root: [u128; 2],
+}
+
+/// Aleo Eth address representation
+#[aleo_serialize]
+#[derive(Debug)]
+pub struct AleoEthAddress {
+    /// Address bytes
+    pub bytes: [u8; 20],
+}
```

Comment:
How do we use this addresses? I wonder if `sender` or `receiver` can have only 20 bytes in their address?